test/unit/org/apache/cassandra/auth/StubAuthorizer.java

```diff
@@ -98,11 +98,19 @@ public void revokeAllFrom(RoleResource revokee)
                 userPermissions.remove(key);
     }
 
-    public void revokeAllOn(IResource droppedResource)
+    public Set<RoleResource> revokeAllOn(IResource droppedResource)
     {
+        Set<RoleResource> roles = userPermissions.keySet()
+                                                 .stream()
+                                                 .filter(key -> key.right.equals(droppedResource))
+                                                 .map(key -> RoleResource.role(key.left))
+                                                 .collect(Collectors.toSet());
+
         for (Pair<String, IResource> key : userPermissions.keySet())
             if (key.right.equals(droppedResource))
                 userPermissions.remove(key);
+        
+        return roles;
```

Comment:
It looks fishy that we compute something, and use it only as return value.
I.e. the computed thing doesn't really need to be in sync with the function logic.

Wouldn't adding to a list just by `userPermissions.remove` be better?